test/show-heartbeat.js

```diff
@@ -0,0 +1,159 @@
+import {expect} from 'chai';
+import jsdom from 'mocha-jsdom';
+
+import {mockNormandy} from './utils';
+import ShowHeartbeatAction from '../actions/show-heartbeat/index';
+
+
+function surveyFactory(props={}) {
+    return Object.assign({
+        message: 'test message',
+        thanksMessage: 'thanks!',
+        postAnswerUrl: 'http://example.com',
+        learnMoreMessage: 'Learn More',
+        learnMoreUrl: 'http://example.com',
+        weight: 100,
+    }, props);
+}
+
+
+function recipeFactory(props={}) {
+    return Object.assign({
+        id: 1,
+        args: {
+            defaults: {},
+            surveys: [surveyFactory()],
+        },
+    }, props);
+}
+
+
+describe('ShowHeartbeatAction', function() {
+    jsdom();
+
+    beforeEach(function() {
+        this.normandy = mockNormandy();
+    });
+
+    it('should run without errors', async function() {
+        let action = new ShowHeartbeatAction(this.normandy, recipeFactory());
+        await action.execute();
+    });
+
+    it('should not show heartbeat if it has shown within the past 7 days', async function() {
+        let recipe = recipeFactory();
+        let action = new ShowHeartbeatAction(this.normandy, recipe);
+
+        this.normandy.mock.storage.data['lastShown'] = '100';
+        this.sinon.stub(Date, 'now').returns(10);
+
+        await action.execute();
+        expect(this.normandy.showHeartbeat).to.not.have.been.called;
+    });
+
+    it('should show heartbeat if it hasn\'t shown within the past 7 days', async function() {
```

Comment:
Nit: Strings with single quotes in them should use double quotes instead of escaping single quotes, where possible.